### PR TITLE
Fix local test error

### PR DESCRIPTION
**Pull request by CunliangGeng**
Please review PR #101 and #102 first, then this PR (only need to review the last two commits).

The test errors of `test_nplinker.py` is caused by `bgc.strain` and `gcf.id`.
This PR fixes the errors with a temporary solution (i.e. using the logics of the original codebase) to keep focusing on the refactoring of the data preparation part.
Added two issue #103 and #104 to update/change the logics later.

### CODE REVIEW

**Comment by CunliangGeng** [comment]
@hechth  Conflicts solved and ready for review

File: src/nplinker/genomics/mibig/mibig_loader.py [hunk 1]
```diff
@@ -108,6 +110,7 @@ def parse_bgc_metadata_json(file: str) -> BGC:
     metadata = MibigMetadata(file)
     mibig_bgc = BGC(metadata.mibig_accession, metadata.biosyn_class)
     mibig_bgc.mibig_bgc_class = metadata.biosyn_class
+    mibig_bgc.strain = Strain(metadata.mibig_accession)
```
Comment (hechth) [inline src/nplinker/genomics/mibig/mibig_loader.py:113]: So the mibig accession is equal to the strain? What info does the mibig accession normally contain?

File: src/nplinker/scoring/linking/link_finder.py [hunk 2]
```diff
@@ -442,8 +446,8 @@ def get_links(self,
             link_levels = [0, 1]
 
             # Get necessary ids
-            # CG: TODO update the logics here
-            #   integer gcf.id has been removed, use string gcf.gcf_id instead.
+            # CG: TODO update the logics here:
+            #   don't use integer gcf.id, use string gcf.gcf_id instead.
             input_ids = np.array([gcf.id for gcf in input_object],
                                  dtype=np.int32)
```
Comment (hechth) [inline src/nplinker/scoring/linking/link_finder.py:452]: Should this todo be adressed immediately or do we plan to take care of this later?

File: src/nplinker/genomics/mibig/mibig_loader.py (hunk 1 above)
Comment (CunliangGeng) [reply src/nplinker/genomics/mibig/mibig_loader.py:113]: No, mibig accession (a string identifier) is not equal to strain, but nplinker uses it as strain. Actually just as what I said this Monday, nplinker does not care about the real name of strain but only need a unique name or label for that. mibig accession is unique, so it's used as strain name. 
Let's keep it this way until we refactor the strain part.

File: src/nplinker/scoring/linking/link_finder.py (hunk 2 above)
Comment (CunliangGeng) [reply src/nplinker/scoring/linking/link_finder.py:452]: Thought about that, better later to fix it.

**Review by hechth — APPROVED** [review]

**Comment by hechth** [comment]
@CunliangGeng let's merge this and move on with the refactoring :)

**Pull request merged**